### PR TITLE
Git command line exec to be applied to ungit

### PR DESCRIPTION
**Pull request by codingtwinky**
cc: @Ajedi32 @brettswift

This will fix the git checkout and I think that was the main concern of the issue #283.  Is there any other operations I should consider?

0dzz2stlbo

### CODE REVIEW

**Comment by Ajedi32** [comment]
A good test case would be `git branch -f some-existing-branch some_commit` since that only affects a branch pointer and doesn't touch the working directory at all.

**Comment by brettswift** [comment]
will this work on the `git branch` command?  I find one thing that's hard for people to learn about git is that git branch doesn't really create a new  visual branch, just a new pointer..  visually this would be nice to show :)  I'd expect two pointers to the same commit..     

I've used ungit a little bit but not a ton so not sure if this works with a manual refresh either

**Comment by codingtwinky** [comment]
Sorry it took so long... there were some "gotchas" that I had to fix for this one...  Is there more I should test?  

fximzctpdg

**Comment by Ajedi32** [comment]
Hmm. How about `git reset --soft <commit>`? That command changes HEAD without touching the working directory or index, so that might also be a good test case. Maybe also make sure `git tag`, `git tag -d`, `git push`, etc work as expected...

File: components/graph/graph.js [hunk 1]
```diff
@@ -127,10 +146,7 @@ GraphViewModel.prototype.loadNodesFromApi = function(callback) {
 
       self.edges(edges);
       self.nodes(nodes);
-
-      if (nodes.length > 0) {
```
Comment (campersau) [inline components/graph/graph.js]: Why did you remove the `if (nodes.length > 0) {` check? Calling `cy()` on `undefined` will throw.

File: components/graph/graph.js [hunk 2]
```diff
@@ -110,13 +110,32 @@ GraphViewModel.prototype.loadNodesFromApi = function(callback) {
...
     .then(function(log) {
-      var nodes = log.nodes ? log.nodes : [];
+      // set new limit and skip
       self.limit(parseInt(log.limit));
       self.skip(parseInt(log.skip));
-      nodes = self.computeNode(nodes.map(function(logEntry) {
-          return self.getNode(logEntry.sha1, logEntry);
-        }));
-
+      return log.nodes || []
```
Comment (campersau) [inline components/graph/graph.js]: Semicolon.

File: components/graph/graph.js (hunk 1 above)
Comment (codingtwinky) [inline components/graph/graph.js]: Nodes will always have least length of 0 and it now is wrapped around max

File: components/graph/graph.js (hunk 2 above)
Comment (codingtwinky) [inline components/graph/graph.js]: done

File: components/graph/graph.js (hunk 1 above)
Comment (campersau) [inline components/graph/graph.js]: Wrapping with `Math.max` does not prevent it from beeing called.

Try the following in a console:
```js
var nodes = [];
Math.max(nodes[nodes.length - 1].cy(), 0);
```

Comment (codingtwinky) [inline components/graph/graph.js]: Ah, you are right... thanks

**Comment by codingtwinky** [comment]
@Ajedi32 This fix definitely works for the `tag` `reset` and `push` for sure and will be adding that tests later.  

I'm trying to see if I can fix this cherry pick test fail for windows but will see...
